data/txHandler.go

```diff
@@ -640,6 +650,12 @@ func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) net
 		}
 	}
 
+	// rate limit per application in a group. Limiting any app in a group drops the entire message.
+	if handler.appLimiter != nil && handler.appLimiter.shouldDrop(unverifiedTxGroup, rawmsg.Sender.(network.IPAddressable).IPAddr()) {
```

Comment:
We discussed offline, but should probably take only the top 64 bits of an ipv6 address.

Reply:
i agree excellent catch

Reply:
fixed, use either ipv4, or 4 lower bytes if embedded address, or high 8 bytes of ipv6